pkg/webhook/policy.go

```diff
@@ -331,20 +333,58 @@ func (h *validationHandler) validateConfigResource(ctx context.Context, req admi
 	return nil
 }
 
+// acquire tries to acquire the validation semaphore.
+// Blocks until the semaphore has capacity or the context is canceled.
+// Returns true if acquired successfully - the caller is responsible
+// for calling release() when true is returned.
+func (h *validationHandler) acquire(ctx context.Context) bool {
+	if h.semaphore == nil {
+		if h.reporter != nil {
+			_ = h.reporter.ReportPendingValidationRequests(0)
+		}
+		return true
+	}
+	if h.reporter != nil {
+		v := atomic.AddInt64(&h.pending, 1)
+		_ = h.reporter.ReportPendingValidationRequests(v)
+
+		defer func() {
+			v := atomic.AddInt64(&h.pending, -1)
+			_ = h.reporter.ReportPendingValidationRequests(v)
+		}()
+	}
+
+	select {
+	case h.semaphore <- struct{}{}:
+		return true
+	case <-ctx.Done():
+		return false
+	}
+}
+
+func (h *validationHandler) release() {
+	if h.semaphore == nil {
+		return
+	}
+	<-h.semaphore
+}
+
+// pendingValidationRequests returns the number of requests blocked
+// waiting for the validation lock.
+// NOTE: Currently incremented only if a statsReporter has been set.
+func (h *validationHandler) pendingValidationRequests() int64 {
```

Comment:
This metric name appears to overlap with the other metric name

Reply:
oops, was meant to be part of a review... still looking through to see if I'm missing a nuance, but I'm trying to figure out where this is used.

Reply:
nvm, I see that addint64 is mutating the underlying value and returning the result, this is all for the same metric

Reply:
This might me not understanding proper usage of OpenCensus - in Prometheus, the underlying the metric registered into the library also manages the underlying value and the interface for mutating it. Here that appears to be separated. @sozercan any input on this?